strawberry/ext/mypy_plugin.py

```diff
@@ -509,16 +531,14 @@ def collect_attributes(self) -> Optional[List[DataclassAttribute]]:
                 type=sym.type,
             )
 
-            # add support for mypy >= 0.800 without breaking backwards compatibility
-            # https://github.com/python/mypy/pull/9380/file
-            # https://github.com/strawberry-graphql/strawberry/issues/678
-
-            try:
-                attribute = DataclassAttribute(**params)  # type: ignore
-            except TypeError:
+            # Support the addition of `info` in mypy 0.800 and `kw_only` in mypy 0.920
+            # without breaking backwards compatibility.
+            if MypyVersion.VERSION >= Decimal("0.800"):
                 params["info"] = cls.info
-                attribute = DataclassAttribute(**params)  # type: ignore
+            if MypyVersion.VERSION >= Decimal("0.920"):
+                params["kw_only"] = False
```

Comment:
we'll need to change this after we merge: https://github.com/strawberry-graphql/strawberry/pull/1187